test/suite/test_prepare_lookaside01.py

```diff
@@ -40,33 +40,66 @@ class test_prepare_lookaside01(wttest.WiredTigerTestCase):
     def conn_config(self):
         return 'cache_size=50MB'
 
-    def prepare_updates(self, session, uri, value, ds, nrows):
-        # Update a large number of records with prepare transactions using
-        # multiple sessions. we'll hang if lookaside table isn't doing its
-        # thing.
+    def prepare_updates(self, uri, ds, nrows, nsessions, nkeys):
+        # Update a large number of records in their individual transactions.
+        # This will force eviction and start lookaside eviction of committed
+        # updates.
         #
-        # If we do all updates in a single session, then hang will be due to
-        # uncommitted updates, instead of prepared updates.
+        # Follow this by updating a number of records in prepared transactions
+        # under multiple sessions. We'll hang if lookaside table isn't doing its
+        # thing. If we do all updates in a single session, then hang will be due
+        # to uncommitted updates, instead of prepared updates.
         #
-        # If we increase nsessions below, then hang will occur.
-        nsessions = 1
+        # Do another set of updates in that many transactions. This forces the
+        # pages that have been evicted to lookaside to be re-read and brought in
+        # memory. Hence testing if we can read prepared updates from lookaside.
+
+        # Start with setting a stable timestamp to pin history in cache
+        self.conn.set_timestamp('stable_timestamp=' + timestamp_str(1))
+
+        # Commit some updates to get eviction and lookaside fired up
```

Comment:
I am not clear as to why we need committed updates for evicting prepared updates.
my understanding is prepared updates should be treated same as committed updates from eviction point of view. committed updates are evicted with out any prerequisites, similarly prepared updates should be evicted with out any prerequisites, committed updates should not be prerequisite for prepared updates eviction.

Reply:
As discussed offline, with the understanding that we are not necessarily avoiding a stall if cache is full of prepared updates, please review if this comment still holds good

Reply:
If we understand how it is working is better, instead of ignoring it.

Reply:
We discussed this one earlier.
I think what happens is eviction notices the committed updates which it tries to evict. It is able to evict one of them, but unable to populate the eviction queue any further because the rest of them are pinned in memory (these updates being newer than the stable timestamp). This inability to populate the eviction queue repeatedly raises the lookaside score beyond a certain value (I believe 80) which sets the cache in a mode that says, it-is-okay-to-try-lookaside-eviction, fkag "WT_CACHE_EVICT_LOOKASIDE". We need the cache in this state otherwise lookaside is not attempted even though the cache might be full with prepared updates.
To conclude, we need at-least some amount of regular eviction while holding updates pinned to raise lookaside score to be able to lookaside evict the prepared updates.